src/index.js

```diff
@@ -47,9 +48,10 @@ export default class GigwalkAPI {
     constructor(config: GigwalkAPIConfig) {
         this.hostname = config.hostname || 'api.apps.gigwalk.com';
 
-        this.requestQueue = new RequestQueue();
+        const client = axios.create({ baseURL: this.hostname });
+        this.requestQueue = new RequestQueue(client);
 
-        const dispatcher: Dispatcher = (request: AxiosXHRConfig<*>): Promise<*> => this.dispatchRequest(request);
+        const dispatcher = this.requestQueue.add;
```

Comment:
This will still cause all the resources to be required to use the request config interface.  Can we change that so something like `this.client('/v1/versions');` could be used as well?